Src/FluentAssertions/Primitives/StringAssertions.cs

```diff
@@ -1884,15 +1884,16 @@ public AndConstraint<TAssertions> BeNullOrWhiteSpace(string because = "", params
     public AndConstraint<TAssertions> BeUpperCased(string because = "", params object[] becauseArgs)
     {
         Execute.Assertion
-            .ForCondition(Subject?.All(char.IsUpper) == true)
+            .ForCondition(Subject is not null && !Subject.Any(char.IsLower))
             .BecauseOf(because, becauseArgs)
-            .FailWith("Expected all characters in {context:string} to be upper cased{reason}, but found {0}.", Subject);
+            .FailWith("Expected all alpha characters in {context:string} to be upper-case{reason}, but found {0}.", Subject);
 
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
     /// <summary>
-    /// Asserts that all characters in a string are not in upper casing.
+    /// Asserts that all of the cased characters in a string some are not upper-case. That is, the string could not be the result
+    /// of a call to <see cref="string.ToUpperInvariant()"/>.
     /// </summary>
     /// <param name="because">
```

Comment:
🔧 I think repeating the remarks from `BeUpperCased` will make this method easier to understand